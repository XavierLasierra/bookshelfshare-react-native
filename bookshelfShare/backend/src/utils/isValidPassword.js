const bcrypt = require("bcrypt");

function isValidPassword(password) {
  return bcrypt.compareSync(password, this.password);
}

module.exports = { isValidPassword };
