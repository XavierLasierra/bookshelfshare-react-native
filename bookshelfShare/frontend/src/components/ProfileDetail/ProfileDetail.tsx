import React from 'react';
import { TouchableOpacity, Text } from 'react-native';

import BookStackIcon from '../../assets/bookStackIcon.svg';
import ShelfIcon from '../../assets/shelfIcon.svg';
import styles from './profileDetail.styles';

export default function ProfileDetail({ handleBookResultsPage, userBooks, shelves }: any) {
  return (
    <>
      <TouchableOpacity
        style={styles.profileButton}
        onPress={() => handleBookResultsPage('currently reading', userBooks.reading)}
      >
        <BookStackIcon width={40} height={40} />
        <Text style={styles.buttonText}>
          {userBooks.reading.length}
          {' '}
          {userBooks.reading.length === 1 ? 'book' : 'books'}
          {' '}
          currently reading
        </Text>
      </TouchableOpacity>
      <TouchableOpacity
        style={styles.profileButton}
        onPress={() => handleBookResultsPage('read', userBooks.read)}
      >
        <BookStackIcon width={40} height={40} />
        <Text style={styles.buttonText}>
          {userBooks.read.length}
          {' '}
          {userBooks.read.length === 1 ? 'book' : 'books'}
          {' '}
          read
        </Text>
      </TouchableOpacity>
      <TouchableOpacity
        style={styles.profileButton}
        onPress={() => handleBookResultsPage('to read', userBooks.toRead)}
      >
        <BookStackIcon width={40} height={40} />
        <Text style={styles.buttonText}>
          {userBooks.toRead.length}
          {' '}
          {userBooks.toRead.length === 1 ? 'book' : 'books'}
          {' '}
          to read
        </Text>
      </TouchableOpacity>
      <TouchableOpacity
        style={styles.profileButton}
        onPress={() => handleBookResultsPage('whislist', userBooks.wishlist)}
      >
        <BookStackIcon width={40} height={40} />
        <Text style={styles.buttonText}>
          {userBooks.wishlist.length}
          {' '}
          {userBooks.wishlist.length === 1 ? 'book' : 'books'}
          {' '}
          on wishlist
        </Text>
      </TouchableOpacity>
      <TouchableOpacity style={styles.profileButton}>
        <ShelfIcon width={40} height={35} />
        <Text style={styles.buttonText}>
          {shelves.length}
          {' '}
          {shelves.length === 1 ? 'shelf' : 'shelves'}
        </Text>
      </TouchableOpacity>
    </>
  );
}
