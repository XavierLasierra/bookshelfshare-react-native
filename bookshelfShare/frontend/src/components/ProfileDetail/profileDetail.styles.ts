import {StyleSheet} from 'react-native';
import stylesConstants from '../../styles/styles.constants';

export default StyleSheet.create({
  profileButton: {
    flexDirection: 'row',
    alignItems: 'center',
    borderTopWidth: 2,
    borderBottomWidth: 2,
    borderColor: stylesConstants.colors.secondary,
    marginBottom: 15,
    height: 50,
  },
  socialButton: {
    width: 160,
    paddingHorizontal: 10,
  },
  buttonText: {
    fontSize: stylesConstants.fontSize.mid,
    marginLeft: 15,
  },
});
