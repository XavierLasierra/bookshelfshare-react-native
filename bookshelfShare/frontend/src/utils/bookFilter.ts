export function bookFilter(text: string, books: any) {
  return books.reduce((acc: any, book: any) => {
    if (book.title.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.publisher.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.isbn.ISBN13.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.isbn.ISBN10.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.authors.some((author: string) => author.toLowerCase()
      .includes(text.toLowerCase()))) return [...acc, book];

    return acc;
  }, []);
}

export function bookShelfListFilter(text: string, books: any) {
  return books.reduce((acc: any, book: any) => {
    if (book.bookData.title.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.publisher.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.isbn.ISBN13.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.isbn.ISBN10.toLowerCase().includes(text.toLowerCase())) return [...acc, book];
    if (book.bookData.authors.some((author: string) => author.toLowerCase()
      .includes(text.toLowerCase()))) return [...acc, book];

    return acc;
  }, []);
}

export function bookShelfLocationFilter(filterLocation: number[], books: any) {
  return books.filter(({ customInformation: { location } }: any) => JSON.stringify(location)
  === JSON.stringify(filterLocation));
}
