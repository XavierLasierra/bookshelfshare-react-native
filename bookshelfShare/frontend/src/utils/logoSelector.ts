import BookIcon from '../assets/bookIcon.svg';
import SearchIcon from '../assets/searchIcon.svg';
import ProfileIcon from '../assets/profileIcon.svg';

export default function logoSelector(logoName: string) {
  switch (logoName) {
    case 'SearchIcon':
      return SearchIcon;
    case 'BookIcon':
      return BookIcon;
    case 'ProfileIcon':
      return ProfileIcon;
    default:
      return BookIcon;
  }
}
