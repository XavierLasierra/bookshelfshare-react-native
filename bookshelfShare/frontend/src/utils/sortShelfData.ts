export default function sortShelfData(shelf: any) {
  return shelf?.books.reduce((acc: any, book: any) => {
    const foundShelfBox = acc.find(({ location }: any) => JSON.stringify(location)
         === JSON.stringify(book?.customInformation?.location));
    if (foundShelfBox) {
      foundShelfBox.books.push(book.bookData);
    } else {
      acc.push({
        location: book?.customInformation?.location,
        books: [book.bookData]
      });
    }
    return acc;
  }, []);
}
