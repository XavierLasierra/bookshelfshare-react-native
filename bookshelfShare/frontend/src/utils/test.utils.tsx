import React from 'react';
import { render as rtlRender } from '@testing-library/react-native';
import { Provider } from 'react-redux';

import configureStore from '../redux/store';

function render(
  component: any,
  initialState: any
) {
  const store = configureStore(initialState);

  function Wrapper({ children }: any) {
    return (
      <Provider store={store}>
        {children}
      </Provider>
    );
  }
  return rtlRender(component, { wrapper: Wrapper });
}

// re-export everything
export * from '@testing-library/react-native';
// override render method
export { render };
