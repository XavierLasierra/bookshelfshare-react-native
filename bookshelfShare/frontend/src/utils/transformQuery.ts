interface Query {
    isbn?: string,
    inauthor?: string,
    intitle?: string,
    inpublisher?: string
}

export default function transformQuery(query: Query):string {
  const queryEntries = Object.entries(query);
  const url = queryEntries
    .reduce((acc, queryElement, index) => `${acc}${queryElement[0]}=${queryElement[1]}${index < queryEntries.length - 1 ? '+' : ''}`, '');
  return `${url}&key=${process.env.GOOGLE_API_KEY}`;
}
