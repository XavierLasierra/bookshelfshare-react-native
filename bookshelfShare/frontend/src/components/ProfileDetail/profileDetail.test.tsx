import React from 'react';
import ProfileDetail from './ProfileDetail';
import { render, fireEvent } from '../../utils/test.utils';
import userBooksMock from '../../mocks/userBooks.mock';
import shelfMock from '../../mocks/shelf.mock';

describe('Given a ProfileDetail component', () => {
  describe('When it is rendered', () => {
    let handleBookResultsPage: any;
    let screen: any;
    beforeEach(() => {
      handleBookResultsPage = jest.fn();
      screen = render(<ProfileDetail
        handleBookResultsPage={handleBookResultsPage}
        userBooks={userBooksMock}
        shelves={[shelfMock]}
      />);
    });

    test('Then should match the snapshot', () => {
      expect(screen).toMatchSnapshot();
    });

    ['readingButton', 'readButton', 'toReadButton', 'wishlistButton']
      .forEach((testId) => {
        describe(`And you press ${testId}`, () => {
          test('Then handleBookresultsPage should have been called', () => {
            const button = screen.getByTestId(testId);
            fireEvent.press(button);

            expect(handleBookResultsPage).toHaveBeenCalled();
          });
        });
      });
  });
});
