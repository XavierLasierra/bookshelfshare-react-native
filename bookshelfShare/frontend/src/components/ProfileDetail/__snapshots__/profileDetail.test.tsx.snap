// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a ProfileDetail component When it is rendered Then should match the snapshot 1`] = `
Array [
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderBottomWidth": 2,
        "borderColor": "#ECECEC",
        "borderTopWidth": 2,
        "flexDirection": "row",
        "height": 50,
        "marginBottom": 15,
        "opacity": 1,
      }
    }
    testID="readingButton"
  >
    <IconMock
      height={40}
      width={40}
    />
    <Text
      style={
        Object {
          "fontSize": 16,
          "marginLeft": 15,
        }
      }
    >
      1
       
      book
       
      currently reading
    </Text>
  </View>,
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderBottomWidth": 2,
        "borderColor": "#ECECEC",
        "borderTopWidth": 2,
        "flexDirection": "row",
        "height": 50,
        "marginBottom": 15,
        "opacity": 1,
      }
    }
    testID="readButton"
  >
    <IconMock
      height={40}
      width={40}
    />
    <Text
      style={
        Object {
          "fontSize": 16,
          "marginLeft": 15,
        }
      }
    >
      1
       
      book
       
      read
    </Text>
  </View>,
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderBottomWidth": 2,
        "borderColor": "#ECECEC",
        "borderTopWidth": 2,
        "flexDirection": "row",
        "height": 50,
        "marginBottom": 15,
        "opacity": 1,
      }
    }
    testID="toReadButton"
  >
    <IconMock
      height={40}
      width={40}
    />
    <Text
      style={
        Object {
          "fontSize": 16,
          "marginLeft": 15,
        }
      }
    >
      1
       
      book
       
      to read
    </Text>
  </View>,
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderBottomWidth": 2,
        "borderColor": "#ECECEC",
        "borderTopWidth": 2,
        "flexDirection": "row",
        "height": 50,
        "marginBottom": 15,
        "opacity": 1,
      }
    }
    testID="wishlistButton"
  >
    <IconMock
      height={40}
      width={40}
    />
    <Text
      style={
        Object {
          "fontSize": 16,
          "marginLeft": 15,
        }
      }
    >
      1
       
      book
       
      on wishlist
    </Text>
  </View>,
  <View
    accessible={true}
    collapsable={false}
    focusable={false}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "borderBottomWidth": 2,
        "borderColor": "#ECECEC",
        "borderTopWidth": 2,
        "flexDirection": "row",
        "height": 50,
        "marginBottom": 15,
        "opacity": 1,
      }
    }
  >
    <IconMock
      height={35}
      width={40}
    />
    <Text
      style={
        Object {
          "fontSize": 16,
          "marginLeft": 15,
        }
      }
    >
      1
       
      shelf
    </Text>
  </View>,
]
`;
